superdesk/io/mediatopics.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import os
+from json import loads
+from .subjectcodes import get_parent_subjectcode
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MediaTopicsIndex:
+
+    def __init__(self):
+        # A map for mapping from an IPTC subject code to a Media Topic
+        self.subject_media_topic_map = dict()
+        # A map for mapping from a Media Topic code to a subject code.
+        self.media_topic_to_subject_map = dict()
+        # A map that contains all the media topics.
+        self.media_topic_map = dict()
+
+        dirname = os.path.dirname(os.path.realpath(__file__))
+
+        # The media topics json file is available at http://cv.iptc.org/newscodes/mediatopic?lang=x-all
+        # there are also language specific files that could be used.
+        media_topic_codes = os.path.join(dirname, 'data', 'cptall-x-all.json')
+        mediatopics = self._load_codes(media_topic_codes)
+        self.generate_mediatopic_to_subject_map(mediatopics.get('conceptSet'))
+
+    def _load_codes(self, filename):
+        with open(filename, 'rb') as f:
+            codes_bytes = f.read()
+            codes = loads(codes_bytes.decode('UTF-8'))
```

Comment:
could be a bit simpler via `with open(filename, 'rb') as f: codes = json.load(f)`

Reply:
Done